server/resp/src/main/java/org/infinispan/server/resp/RespServer.java

```diff
@@ -145,7 +145,11 @@ public AdvancedCache<byte[], byte[]> getCache() {
    }
 
    public Resp3Handler newHandler() {
-      return new Resp3Handler(this, configuredValueType);
+      return new Resp3Handler(this, configuredValueType, getCache());
+   }
+
+   public Resp3Handler newHandler(AdvancedCache<byte[], byte[]> cache) {
```

Comment:
This method seems needed? Its only usage is to pass in the cache from this Handler, is it not the same as the no arg newHandler method?

Reply:
This one is here only for the AUTH to invoke it after authentication. All the other places still use to other no-arg.

Reply:
Hrmm, all the other places then would drop the secure cache if I am not mistaken as well right?

Reply:
Good point. I've removed the no-arg method.